coordinator/tests/integration/helpers.py

```diff
@@ -232,28 +254,56 @@ def deploy_distributed_cluster(
 
 
 def get_traces(
-    tempo_host: str, service_name="tracegen", tls=True, nonce: Optional[str] = None
+    tempo_host: str,
+    service_name="tracegen",
+    tls=True,
+    nonce: Optional[str] = None,
+    source_pod: Optional[str] = None,
+    juju: Optional[Juju] = None,
```

Comment:
rather than making `get_traces` do two very different things, I'd much rather rename it to `get_traces_from_localhost` and create a new separate `get_traces_from_unit` method that does what you need it to do. 
Same for `get_traces_patiently --> get_traces_patiently_from_localhost`.